src/curvefit/models/core_model.py

```diff
@@ -65,12 +65,18 @@ def get_params(self, x):
             self.data_inputs.var_link_fun,
         )
 
-    def predict(self, x, t, predict_fun=None):
+    def predict(self, x, t, predict_fun=None, is_multi_group=False):
```

Comment:
@jizezhang so is this where if you pass multi_group you'll get all the predictions for every group? how do you know which group is which?

Reply:
so the `params` has shape `num_params-by-num_groups` and the groups would be in that order, which I think is also the order of `data` passed in when fitting

Reply:
great!